zebra-chain/src/block.rs

```diff
@@ -105,3 +133,18 @@ impl TrustedPreallocate for Hash {
         ((MAX_PROTOCOL_MESSAGE_LEN - 1) as u64) / BLOCK_HASH_SIZE
     }
 }
+
+use thiserror::Error;
+
+/// Errors that can occur when checking Block consensus rules.
+///
+/// Each error variant corresponds to a consensus rule, so enumerating
+/// all possible verification failures enumerates the consensus rules we
+/// implement, and ensures that we don't reject blocks or transactions
+/// for a non-enumerated reason.
+#[allow(dead_code, missing_docs)]
+#[derive(Error, Debug, PartialEq)]
+pub enum TransactionError {
```

Comment:
Since this is for `Block`-level consensus rules, shouldn't this be a `BlockError`? It's not about any one transaction, really.

Reply:
Also I might put this in its own `error.rs` in the `block` module, with `thiserror::Error` at the top of its scope (aka, the top of the module file)

Reply:
I had intentions to move the errors into an `error.rs` but forgot to do it, thank you for that.

`BlockError` makes more sense here, thanks, i made the changes at https://github.com/ZcashFoundation/zebra/pull/2354/commits/3c5a52192b366b6627c0205d44f2ecff816ca2b5